src/clients/entitlements/entitlements.user-scoped.ts

```diff
@@ -35,7 +40,22 @@ export class EntitlementsUserScoped<T extends TEntity = TEntity> {
     }
   }
 
-  async isEntitledToFeature(featureKey: string): Promise<IsEntitledResult> {
+  async isEntitledToFeature(featureKey: string, attributes: Record<string, any> = {}): Promise<IsEntitledResult> {
+    const isEntitledResult = await this.getEntitlementResult(featureKey);
+
+    if (!isEntitledResult.result) {
+      const ffResult = await this.getFeatureFlagResult(featureKey, { ...this.predefinedAttributes, ...attributes });
```

Comment:
In the "prepareAttributes" function, I first applied the "custom-attributes" and then the "predefined" ones.

we need to be aligned here, it might be a product-wise question, if we want to allow overriding these predefined attributes.

Reply:
fixed